firefox-ios/Client/TabManagement/Legacy/LegacyTabManager.swift

```diff
@@ -637,15 +637,24 @@ class LegacyTabManager: NSObject, FeatureFlaggable, TabManager, TabEventHandler
     @MainActor
     func removeAllTabs(isPrivateMode: Bool) async {
         let currentModeTabs = tabs.filter { $0.isPrivate == isPrivateMode }
+        var currentSelectedTab: BackupCloseTab?
+
+        // Backup the selected tab in separate variable as the `removeTab` method called below for each tab will
+        // automatically update tab selection as if there was a single tab removal.
         if let tab = selectedTab, tab.isPrivate == isPrivateMode {
-            backupCloseTab = BackupCloseTab(tab: tab,
-                                            restorePosition: tabs.firstIndex(of: tab),
-                                            isSelected: selectedTab?.tabUUID == tab.tabUUID)
+            currentSelectedTab = BackupCloseTab(tab: tab,
+                                                restorePosition: tabs.firstIndex(of: tab),
+                                                isSelected: selectedTab?.tabUUID == tab.tabUUID)
         }
         backupCloseTabs = tabs
+
         for tab in currentModeTabs {
             await self.removeTab(tab.tabUUID)
         }
+
+        // Save the tab state that existed prior to removals (preserves original selected tab)
+        backupCloseTab = currentSelectedTab
```

Comment:
Should we add tests to check for this?

Reply:
Wouldn't be a bad idea! I just pushed up a couple tests.

It took me a while to find the right suite (`TabManagerTests.swift`). There do not appear to be many tab manager implementation tests as it is. Or did I miss a file as I was looking? 🤔

Reply:
Correct, there are not a lot of Tab Manager Implementation tests 😭 thanks for adding some, and thanks @cyndichin for advocating for it!

Reply:
great thank you so much for writing up the tests so quickly!